meinberlin/apps/dashboard2/contents.py

```diff
@@ -1,25 +1,37 @@
-class DashboardContents:
-    _registry = {'projects': {}, 'modules': {}}
+def _component_sort_key(component):
+    return component.weight, component.identifier
 
-    @property
-    def projects(self):
-        return self._registry['projects']
 
-    @property
-    def modules(self):
-        return self._registry['modules']
+class DashboardContents:
+    _registry = {'projects': {}, 'modules': {}}
 
     def register_project(self, component):
-        self._registry['projects'][component.identifier] = component
+        self._register('projects', component)
 
     def register_module(self, component):
-        self._registry['modules'][component.identifier] = component
+        self._register('modules', component)
+
+    def _register(self, section, component):
+        if component.identifier in self._registry[section]:
+            raise ValueError('Identifier ({}) is already registered'
+                             .format(component.identifier))
+        self._registry[section][component.identifier] = component
+
+    def unregister_project(self, component):
+        identifier = getattr(component, 'identifier', component)
+        self._registry['projects'].pop(identifier, None)
+
+    def unregister_module(self, component):
+        identifier = getattr(component, 'identifier', component)
+        self._registry['modules'].pop(identifier, None)
```

Comment:
i removed them until we need them. reverting a single commit will be fairly easy